website/addons/base/testing/models.py

```diff
@@ -0,0 +1,392 @@
+
+    def test_set_user_auth(self):
+        node_settings = self.NodeSettingsFactory()
+        user_settings = self.UserSettingsFactory()
+        external_account = self.ExternalAccountFactory()
+
+        user_settings.owner.external_accounts.append(external_account)
+        user_settings.save()
+
+        node_settings.external_account = external_account
+        node_settings.set_auth(external_account, user_settings.owner)
+        node_settings.save()
+
+        assert_true(node_settings.has_auth)
+        assert_equal(node_settings.user_settings, user_settings)
+        # A log was saved
+        last_log = node_settings.owner.logs[-1]
+        assert_equal(last_log.action, '{0}_node_authorized'.format(self.short_name))
+        log_params = last_log.params
+        assert_equal(log_params['node'], node_settings.owner._primary_key)
+        assert_equal(last_log.user, user_settings.owner)
+
+    def test_serialize_credentials(self):
+        self.user_settings.external_accounts[0].oauth_key = 'key-11'
+        self.user_settings.save()
+        credentials = self.node_settings.serialize_waterbutler_credentials()
+
+        expected = {'token': self.node_settings.external_account.oauth_key}
+        assert_equal(credentials, expected)
+
+    def test_serialize_credentials_not_authorized(self):
+        self.node_settings.user_settings = None
+        self.node_settings.save()
+        with assert_raises(exceptions.AddonError):
+            self.node_settings.serialize_waterbutler_credentials()
+
+    def test_serialize_settings(self):
+        settings = self.node_settings.serialize_waterbutler_settings()
+        expected = {'folder': self.node_settings.folder_id}
+        assert_equal(settings, expected)
+
+    def test_serialize_settings_not_configured(self):
+        self.node_settings.clear_settings()
+        self.node_settings.save()
+        with assert_raises(exceptions.AddonError):
+            self.node_settings.serialize_waterbutler_settings()
+
+    def test_create_log(self):
+        action = 'file_added'
+        path = 'pizza.nii'
+        nlog = len(self.node.logs)
+        self.node_settings.create_waterbutler_log(
+            auth=Auth(user=self.user),
+            action=action,
+            metadata={'path': path, 'materialized': path},
+        )
+        self.node.reload()
+        assert_equal(len(self.node.logs), nlog + 1)
+        assert_equal(
+            self.node.logs[-1].action,
+            '{0}_{1}'.format(self.short_name, action),
+        )
+        assert_equal(
+            self.node.logs[-1].params['path'],
+            path
+        )
+
+    def test_after_fork_by_authorized_user(self):
+        fork = ProjectFactory()
+        clone, message = self.node_settings.after_fork(
+            node=self.node, fork=fork, user=self.user_settings.owner
+        )
+        assert_equal(clone.user_settings, self.user_settings)
+
+    def test_after_fork_by_unauthorized_box_user(self):
```

Comment:
Since this test is part of a mixin used by several addons, I think it should be named `test_after_fork_by_unauthorized_user`